### ISSUE TITLE
Expose VimState for extensibility

### ISSUE THREAD

**Issue by lloeki** [atom/vim-mode#536]
Having read access to an editor's `VimState` is bound to be very useful for third-partyvim-mode extensions such as [ex-mode](https://github.com/lloeki/ex-mode). Currently, this is completely prevented, even in the most hackish ways, because the references are unreachable, as they're trapped into a closure scope:

```
--- lib/vim-mode.coffee.orig	2015-02-23 11:04:48.000000000 +0100
+++ lib/vim-mode.coffee	2015-02-23 11:04:58.000000000 +0100
@@ -9,9 +9,9 @@
 
   activate: (state) ->
     @disposables = new CompositeDisposable
-    globalVimState = new GlobalVimState
+    @globalVimState = new GlobalVimState
     @statusBarManager = new StatusBarManager
-    vimStates = new WeakMap
+    @vimStates = new WeakMap
 
     @disposables.add atom.workspace.observeTextEditors (editor) =>
       return if editor.mini
```

Subsequently, `ex-mode` will be able to query an editor's VimState on its own:

```
VimMode = atom.packages.activePackages['vim-mode'].mainModule
VimMode.globalVimState
VimMode.vimStates.get(atom.workspace.getActiveEditor())
```

Alternatives such as implementing accessor methods (such as `VimMode.getGlobalState()` and `VimMode.getVimState(editor)`, possibly cloning the state objects to guarantee read-only-ness) or going full steam ahead and providing a versioned service, like [atom/status-bar](https://github.com/atom/status-bar/blob/master/package.json#L16) are equally welcome.

**Comment by maxbrunsfeld** [comment]
Yeah, we need to provide a versioned service. I'd be open to a PR that adds one if you're interested in working on that.

**Comment by lloeki** [comment]
I'm on it.

**Comment by maxbrunsfeld** [comment]
Closed via #540.
